README.md

```diff
@@ -49,33 +49,92 @@ Relinquish admin rights...
+This project includes the following smart contracts and their metadata:
+- [x] [Smart Contracts](./src)
+  - [x] Redistribution
+  - [x] Staking Registry
+  - [x] Price Oracle
+  - [x] Postage Stamps
+  - [x] Test Token
+
+- [x] Metadata ([Testnet](./testnet_deployed.json),[Mainnet](./mainnet_deployed.json))
+  - [x] **Chain ID**: Chain ID of the blockchain.
+  - [x] **Network ID**: Network ID.
+  - [x] **ABI**: Interface to communicate with smart contracts.
+  - [x] **Bytecode**: Compiled object code that is executed during communication with smart contract.
+  - [x] **Address**: Address of the deployed contract on blockchcain.
+  - [x] **Block**: Block height in which the transaction is mined.
+  - [x] **URL**: URL for analyzing the transaction.
+
+### [Scripts](./scripts)
+- [x] Script for deploying all and individual contracts
+- [ ] Script to deploy contracts and run bee node with updated addresses
+- [ ] Script for interacting with smart contracts
+- [x] Script assigning roles/permissions for smart contracts
+  - [x] Redistributor roles
+
+## Project Setup
+### Prerequisites
 To set up the project, you will need `yarn` and `node`.
 
 The project has been tested with the latest node LTS (Erbium). A `.nvmrc` file is also provided.
 
-## Installation:
-
-Run `yarn install` to install all depencencies.
-
-## Testing:
+### Setup
+To get started with this project, follow these steps:
 
-You may run tests with `yarn test`.
-
-Hardhat is configured to deploy all contracts to the testing hardhat devchain and to use all named accounts.
+1. Clone the repo.
+2. Run `yarn install` at the root of the repo to install all dependencies.
+3. Add a `.env` file in your root directory, where you'll store your sensitive information for deployment. An example file [`.env.example`](./.env.example) is provided for reference.
 
 ## Run
 
-To deploy the smart contracts on Testnet, execute the following command `npx hardhat run scripts/deployer.ts --network testnet`.
-
-To deploy the smart contracts on Mainnet, execute the following command `npx hardhat run scripts/deployer.ts --network mainnet`.
-
-To deploy the smart contracts on Localhost, Geth node or Ganache, execute the following command `npx hardhat run scripts/deployer.ts --network "NETWORK""`.
+### [Tests](./test)
+- [x] Unit Tests
+  - Run `npm run test` to run all the tests.
+  - Run `npm run test:coverage` to see the coverage of smart contracts.
+- [ ] Integration Tests
+
+
+### Deployments
+
+#### Remote
+##### Prerequisites
+Consult devops/storage team for infura token or create one from [Infura website](https://infura.io/).
+
+##### Steps
+1. Run `npm run compile` to get all the contracts compiled.
+2. Run `npm run test` to run all the tests.
+3. Configure `.env` file
+  - Add your wallet `Mnemonic` in `.env` file.
+  - Add your `Infura` token in `.env` file.
+4. To deploy all contracts and set roles:
+  - Mainnet: `npm run deploy:mainnet`
+  - Testnet: `npm run deploy:testnet`
+
```

Comment:
I'd add a note here that after successfully deploying to mainnet or testnet the `mainnet_deployed.json` and `testnet_deployed.json` will be automatically updated and those changes should be committed.